lib/gtfs/timepoint.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Gtfs.Timepoint do
+  alias Gtfs.Csv
+
+  @type id :: String.t()
+
+  @type t :: %__MODULE__{
+          id: id(),
+          name: String.t() | nil
+        }
+
+  @type timepoints_by_id :: %{id() => t()}
+
+  @enforce_keys [
+    :id
+  ]
+
+  @derive Jason.Encoder
+
+  defstruct [
+    :id,
+    name: nil
+  ]
+
+  @spec from_csv_row(Csv.row()) :: t()
+  def from_csv_row(row) do
+    %__MODULE__{
+      id: row["checkpoint_id"],
+      name: row["checkpoint_name"]
+    }
+  end
+
+  @spec timepoint_for_id(timepoints_by_id(), id()) :: t() | nil
```

Comment:
In `data.ex`, if you filter out the nil timepoint_ids before looking up them instead of after, `Timepoint.timepoint_for_id` wouldn't need the nil case here. It could accept an always present `id()` and return an always present `t()`.

Otherwise, the argument should be `id() | nil`.

Reply:
Yep, that's better.